DICOM/src/Microsoft.Health.Anonymizer.Common.UnitTests/PerturbTests.cs

```diff
@@ -0,0 +1,251 @@
+        {
+            yield return new object[] { new AgeObject(10, AgeType.Day), new PerturbSetting() { Span = 1, RoundTo = 0, RangeType = PerturbRangeType.Proportional }, 5, 15 };
+            yield return new object[] { new AgeObject(50, AgeType.Month), new PerturbSetting() { Span = 20, RoundTo = 0, RangeType = PerturbRangeType.Fixed }, 40, 60 };
+            yield return new object[] { new AgeObject(25, AgeType.Year), new PerturbSetting() { Span = 2, RoundTo = 0, RangeType = PerturbRangeType.Proportional }, 0, 50 };
+        }
+
+        public static IEnumerable<object[]> GetShortIntegerToPerturb()
+        {
+            yield return new object[] { (short)100, new PerturbSetting() { Span = 100, RoundTo = 0, RangeType = PerturbRangeType.Fixed }, (short)50, (short)150 };
+            yield return new object[] { (short)0, new PerturbSetting() { Span = 4, RoundTo = 1, RangeType = PerturbRangeType.Fixed }, (short)-2, (short)2 };
+            yield return new object[] { (short)5e2, new PerturbSetting() { Span = 0.5, RoundTo = 2, RangeType = PerturbRangeType.Proportional }, (short)375, (short)625 };
+        }
+
+        public static IEnumerable<object[]> GetUnsignedShortIntegerToPerturb()
+        {
+            yield return new object[] { (ushort)5, new PerturbSetting() { Span = 0, RoundTo = 0, RangeType = PerturbRangeType.Fixed }, (ushort)5, (ushort)5 };
+            yield return new object[] { (ushort)0, new PerturbSetting() { Span = 4, RoundTo = 1, RangeType = PerturbRangeType.Fixed }, (ushort)0, (ushort)2 };
+            yield return new object[] { (ushort)5e2, new PerturbSetting() { Span = 100, RoundTo = 2, RangeType = PerturbRangeType.Fixed }, (ushort)450, (ushort)550 };
+        }
+
+        public static IEnumerable<object[]> GetLongToPerturb()
+        {
+            yield return new object[] { (long)int.MaxValue, new PerturbSetting() { Span = 1, RoundTo = 0, RangeType = PerturbRangeType.Proportional }, (long)int.MaxValue / 2, (long)int.MaxValue / 2 * 3 };
+            yield return new object[] { 10L, new PerturbSetting() { Span = 4, RoundTo = 1, RangeType = PerturbRangeType.Fixed }, 8L, 12L };
+            yield return new object[] { (long)5e2, new PerturbSetting() { Span = 100, RoundTo = 2, RangeType = PerturbRangeType.Fixed }, 450L, 550L };
+        }
+
+        public static IEnumerable<object[]> GetUnsignedLongToPerturb()
+        {
+            yield return new object[] { (ulong)int.MaxValue, new PerturbSetting() { Span = 1, RoundTo = 0, RangeType = PerturbRangeType.Proportional }, (long)int.MaxValue / 2, (long)int.MaxValue / 2 * 3 };
+            yield return new object[] { 10UL, new PerturbSetting() { Span = 100, RoundTo = 1, RangeType = PerturbRangeType.Fixed }, 0UL, 60UL };
+            yield return new object[] { (ulong)5e2, new PerturbSetting() { Span = 100, RoundTo = 2, RangeType = PerturbRangeType.Fixed }, 450UL, 550UL };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetStringValueToPerturb))]
```

Comment:
🔴 [Insist]

Add tests for the null string or the string which cannot be parsed to decimals